### PR TITLE
Tool runtime support

### PR DESCRIPTION
**Pull request by weshaggard**
Add support for running tools on top of .NET Core, things like GenFacades.

This change will restore a version of the Console .NET Core meta-package and any other dependencies into a shared directory, $(ToolRuntimePath), also including a host set via $(ToolHost).

Right now this only works on windows because we don't have all the necessary packages for x-plat yet. I'm actively working to get those enabled.

### CODE REVIEW

**Comment by weshaggard** [comment]
@ellismg @joperezr I address your feedback but given I rebased and you made comments on the commits the comments got pushed aside. See https://github.com/weshaggard/buildtools/commit/4a5729faf95ae4eb925e152b003a9247dfc614a2#commitcomment-13217758 for my follow-up comments.

cc @mellinoe as FYI

I'm doing some more testing from the corefx repo with these changes and then I will merge and update the build tools version.

File: src/nuget/Microsoft.DotNet.BuildTools.nuspec [hunk 1]
```diff
@@ -41,6 +43,8 @@
     <file src="Microsoft.DotNet.Build.Tasks\NativeBinaries\amd64\git2-69db893.dll" target="lib\NativeBinaries\amd64" />
     <file src="Microsoft.DotNet.Build.Tasks\NativeBinaries\x86\git2-69db893.dll" target="lib\NativeBinaries\x86" />
     <file src="Microsoft.DotNet.Build.Tasks\AssemblyInfoPartial.*" target="lib" />
+    <file src="GenFacades\GenFacades.dll" target="lib" />
```
Comment (mellinoe) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: If we plan to use this with CoreRun, we can probably stop building it with CoreConsole and having it rename the extension from exe -> dll. There's a post-step in the publishexe.targets file that renames the file if it detects CoreConsole.exe next to the project output.

**Comment by mellinoe** [comment]
LGTM

File: src/nuget/Microsoft.DotNet.BuildTools.nuspec (hunk 1 above)
Comment (weshaggard) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: I do plan to clean-up the GenFacades projects and also delete their packages. However I'm going to do that in a separate PR.

Comment (joperezr) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: I can do that once I checkin my stuff with BclRewriter since I'll have to do it as well as I'm also using CoreConsole.

Comment (weshaggard) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: I just cleaned it up, see https://github.com/dotnet/buildtools/pull/263
